### DISCUSSION TITLE
My humble experiments with NiceGUI: ui.stepper, styling and gettext

### DISCUSSION THREAD

**Discussion by frica** [zauberzeug/nicegui#2883] [Show and tell]
Hello,

I discovered NiceGUI a month ago and so far I'm amazed by how easy it is to do some simple stuff that looks good.
I made 2 "dummy" apps: 
* My [first one](https://github.com/frica/arewebro) was intended to test the standalone/native aspect (and I managed to make it work, after some sweat)
* The second one is basically a cooking recipe (don't shoot me, people with professional apps using Redis and all), based on the `ui.stepper` control.

It was an opportunity to play with the various styling options. It was [not always easy for me](https://frica.github.io/blog/friends-challenge-4-lasagnapp-or-my-styling-experiments-with-nicegui.html) as I am not a developer nor a frontend guy but I do appreciate the flexibility and the rendering.

I could also test if NiceGUI is compatible with `gettext` (could not see any explicit mention of it in the documentation) and, probably not a surprise for the maintainers but **it works** :) ! I could also make the page refresh after changing the language which is nice.

My only remaining issue is I can't refresh the translation of the `ui.label` in the `ui.header`. 
Also NiceGUI tells me in the console that a ui.header can't be in a @ui.refreshable method as of NiceGUI 1.5. 
So I'm guessing there is maybe some limitation regarding the header with NiceGUI or in general.

The app (if I dare call it like that) can be seen [here](https://lasagnapp.fly.dev/) (repo [there](https://github.com/frica/Lasagnapp)).

Thanks for the framework to all people involved, it's a lot nicer to learn some Python when the result looks decent with so little efforts!

**Comment by falkoschindler** [comment]
Thanks for the warm words, @frica!
We're glad to hear that you like NiceGUI. 🙂

Regarding the `ui.header`: There has been confusion about layout elements like `ui.header` being nested within other elements, which is not its intended use and can lead to layout issues. Therefore we added this warning. But `ui.refreshable` automatically wraps its content into a container which it can clear for rerendering. In order to avoid the warning you're seeing, I recommend placing the refreshable with all its dynamic content _inside_ `ui.header` instead of the other way around.
